### ISSUE TITLE
[BUG] js_of_ocaml hangs on mutually recursive functions

### ISSUE THREAD

**Issue by robz** [ocsigen/js_of_ocaml#1042]
**Describe the bug**
js_of_ocaml seems to hang when compiling this:

```
let rec f1 = () => f1()
and f2 = () => f1();
```

**Expected behavior**
I would expect it to compile successfully since it successfully compiles these cases:

```
let rec f1 = () => f2()
and f2 = () => f1();

let rec g = () => g();
```

**Versions**
Run "esy" in this repo: https://github.com/robz/js_of_ocaml-recursion-bug-repro
See the esy.lock file for package versions

**Comment by hhugo** [comment]
Thanks for the report.
There is a fix in #1043

**Comment by robz** [comment]
Wow, thanks for the fast fix! I see the issue doesn't repro if the --noinline flag is passed, so that also unblocks me.
